### ISSUE TITLE
code question

### ISSUE THREAD

**Issue by Ainccci** [pwang322/DLF#13]
Dear Author,
Why does deleting layers that are defined but not used affect the reproducibility of the code?
For example, delete：
self.proj_cosine_l = nn.Linear(combined_dim_low * (self.len_l - args.conv1d_kernel_size_l + 1), combined_dim_low)     
self.proj_cosine_v = nn.Linear(combined_dim_low * (self.len_v - args.conv1d_kernel_size_v + 1), combined_dim_low)
self.proj_cosine_a = nn.Linear(combined_dim_low * (self.len_a - args.conv1d_kernel_size_a + 1), combined_dim_low)
self.trans_a_with_l = self.get_network(self_type='al')
self.trans_a_with_v = self.get_network(self_type='av')
self.trans_v_with_l = self.get_network(self_type='vl')
self.trans_v_with_a = self.get_network(self_type='va')
model_distill_homo = getattr(get_distillation_kernel_homo, 'DistillationKernel')(n_classes=1,
                                                                               hidden_size=
                                                                               args.dst_feature_dim_nheads[0],
                                                                               gd_size=args.gd_size_low,
                                                                               to_idx=to_idx, from_idx=from_idx,
                                                                               gd_prior=softmax([0, 0, 1, 0, 1, 0], 0.25),
                                                                               gd_reg=10,
                                                                               w_losses=args.w_losses_low,
                                                                               metric=args.metric_low,
                                                                               alpha=1 / 8,
                                                                               hyp_params=args)

        model_distill_hetero = getattr(get_distillation_kernel, 'DistillationKernel')(n_classes=1,
                                                                                   hidden_size=
                                                                                   args.dst_feature_dim_nheads[0] * 2,
                                                                                   gd_size=args.gd_size_high,
                                                                                   to_idx=to_idx, from_idx=from_idx,
                                                                                   gd_prior=softmax([0, 0, 1, 0, 1, 1], 0.25),
                                                                                   gd_reg=10,
                                                                                   w_losses=args.w_losses_high,
                                                                                   metric=args.metric_high,
                                                                                   alpha=1 / 8,
                                                                                   hyp_params=args)

**Comment by peter-wang321** [comment]
Dear Guys,

It is a little confusing. But I guess if it is part of the architecture, I can assume it might affect the statistical property of the computation in PyTorch. You are right, it is not used in the paper, and I am also not sure what the reason is. If you find the real reason, please keep me posted.  Thanks in advance.

Anyway, thanks for your interest in our work.
